### ISSUE TITLE
bug: Window focus wraps around, moving west from leftmost window on 90° rotated display

### ISSUE THREAD

**Issue by torgeir** [koekeishiya/yabai#1511]
With a dual screen setup like this

```
+----------++-------+
| a        || b     |
|          ||       |
+----------+|       |
            |       |
            |       |
            +-------+
```

The following command, executed from the rightmost window in `a`,  exits with code 1

```
❯ yabai -m window --focus east
could not locate a eastward managed window.
❯ echo $?
1
```

..making it a great candidate for chaining, e.g. like this `yabai -m window --focus east  || yabai -m display --focus east` to move focus to display `b`

However, the opposite, ran from the leftmost window on the 90° rotated display `b` exits with code 0

```
❯ yabai -m window --focus west
❯ echo $?
0
```

..and instead "wraps" the focus to the last window on display `b`.

Edit:

This is on `yabai-v5.0.1`

**Comment by torgeir** [comment]
I am not experiencing this any longer with yabai-v5.0.2.

**Comment by torgeir** [comment]
Either this has returned or I misunderstood the details of what was failing the last time around, cause you will not experience the following if you only have one window on each of the displays.

Here's some more details:

With displays `a` and `b`, `b` rotated 90°, and windows `w` and `w' in a vertical split`;
Starting with the selected window `w'`, using the key binding `alt-h` moves `w'` to `w` to `a`, like I would expect.

```
a            b
+----------++-------+
|          || w | w'|
|          ||   |   |
+----------+|   |   |
            |   |   |
            |   |   |
            +-------+
```

With displays `a` and `b`, `b` rotated 90°, and windows `w` and `w' in a horizontal split`;
Starting with the selected window `w'`, using the key binding `alt-h` moves `w'` to `w` and continues cycling `w'` and `w`, never reaching monitor a, this was unexpected to me.

```
a            b
+----------++-------+
|          || w     |
|          ||       |
+----------++-------+
            | w'    |
            |       |
            +-------+
```

 I would expect the movement to go directly from `w'` to display `a`
            
This is yabai-v5.0.4

**Comment by slam** [comment]
I experienced a similar issue, and I hope my workaround can help you as well.

Upon briefly examining the code, I suspected that the slight overlapping of windows was triggering a bug related to window focus management. By introducing a small gap between the windows, my specific issue was resolved.

I set my window gap to 5:

```
yabai -m config window_gap 5
```

**Comment by koekeishiya** [comment]
I was unable to reproduce this specific issue, but I have fixed an off-by-one issue on master that caused a similar issue: https://github.com/koekeishiya/yabai/issues/1463#issuecomment-1424986203

**Comment by koekeishiya** [comment]
Probably fixed in v5.0.5. Feel free to reopen if you can reproduce the issue in the new version.

**Comment by torgeir** [comment]
Works like a charm, thanks!
